pkg/kubelet/kubelet.go

```diff
@@ -2773,8 +2773,17 @@ func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, *v1.Pod, v1.PodResizeStatus)
 	}
 
 	if ok, failReason, failMessage := kl.canAdmitPod(otherActivePods, podCopy); !ok {
-		// Log reason and return. Let the next sync iteration retry the resize
+		// Log reason and return. Let the next sync iteration retry the resize.
+		// If cpu manager policy is static, set resize status to infeasible.
 		klog.V(3).InfoS("Resize cannot be accommodated", "pod", podCopy.Name, "reason", failReason, "message", failMessage)
+		if utilfeature.DefaultFeatureGate.Enabled(features.CPUManager) {
+			if utilfeature.DefaultFeatureGate.Enabled(features.CPUManagerPolicyOptions) {
+				if kl.GetConfiguration().CPUManagerPolicy == "static" {
+					kl.updateContainerResourceAllocation(podCopy)
+					return false, podCopy, v1.PodResizeStatusInfeasible
```

Comment:
I think it would be better to bubble up errors instead of reconstructing the configuration which lead to this. This will require a possibly large refactoring though, but it's so much better.
Besides, the CPUManager feature gate is removed, and the CPUManagerPolicyOptions feature gate is not relevant for the basic configuration here (e.g. doesn't drive the policy)

Reply:
Thanks, acknowledged i think i understand what you mean.

Reply:
Did you mean something like that?
https://github.com/kubernetes/kubernetes/blob/657c2f9f793d64405e4a3fddd440b6d70dd70256/pkg/kubelet/kubelet.go#L2781-L2786

Please note the nil is important because this ensures that pod status will not change on error.
https://github.com/kubernetes/kubernetes/blob/657c2f9f793d64405e4a3fddd440b6d70dd70256/pkg/kubelet/kubelet.go#L2830-L2833